### PR TITLE
Total hours recalculation fix

### LINKED ISSUE #243: Write test for hours summary
Scenarios:
- Add new tracking.
- Change existing tracking
- Change work type on existing tracking

### PR DESCRIPTION
**Pull request by guytepper**
#243

### CODE REVIEW

File: behat/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -186,5 +192,109 @@ private function waitForXpathNode($xpath, $appear = TRUE) {
...
+  public function iAddHoursToLatestTrackingEntry($hours) {
+    $entryID = $this->getLatestTrackingEntry();
+    $this->getSession()->visit($this->locatePath('node/' . $entryID .'/edit'));
+    $element = $this->getSession()->getPage();
+
+    $timeSpentElm = $element->find('css', '#edit-field-issues-logs-und-0-field-time-spent-und-0-value');
+    $timeSpent = $timeSpentElm->getValue();
+
+    if ( isset($timeSpent) ) {
+      $hours +=$timeSpent;
```
Comment (bricel) [inline behat/features/bootstrap/FeatureContext.php]: $hours +=$timeSpent; => $hours += $timeSpent;

File: productivity/modules/custom/productivity_migrate/includes/ProductivityMigrateGithubIssue.inc [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+   * @var array
+   */
+  public $csvColumns = array(
+    array('id', 'ID'),
+    array('title', 'Title'),
+    array('body', 'Description'),
+    array('field_github_project_id', 'GitHub Project ID'),
+    array('field_issue_id', 'Issue ID'),
+    array('field_push_date', 'Push Date'),
+    array('field_employee	', 'Employee'),
```
Comment (bricel) [inline productivity/modules/custom/productivity_migrate/includes/ProductivityMigrateGithubIssue.inc]: array('field_employee	', 'Employee'),
 =>
 array('field_employee', 'Employee'),

File: productivity/modules/custom/productivity_time_tracking/productivity_time_tracking.module [hunk 3]
```diff
@@ -514,6 +514,17 @@ function productivity_time_tracking_bulk_update($month, $year, $employee, $proje
 }
 
 /**
+ * Implements hook_node_insert().
+ */
+function productivity_time_tracking_node_insert($node) {
```
Comment (bricel) [inline productivity/modules/custom/productivity_time_tracking/productivity_time_tracking.module]: Do we need this new hook?

**Comment by nader77** [comment]
Hi @guytepper, I'll have a look on the tests for this PR, 
What's the status here ? Only need to fix the tests ?

**Comment by bricel** [comment]
@nader77 - I already reviewed, @guytepper said that he's able to run the test locally, but I wasn't.

**Comment by nader77** [comment]
@bricel Could you please cherry pick this commit as well: https://github.com/Gizra/productivity/pull/247/commits/981354adb2ba93de2bf5ebde6b2a25e07fde1faa

This will remove hours from a project when a tracking entity is removed through the app.

The tests needs some work still, Making a lot of trouble.

**Comment by nader77** [comment]
@bricel Ready for code review.

**Pull request merged**